core/io/marshalls.cpp

```diff
@@ -59,6 +62,13 @@ ObjectID EncodedObjectAsID::get_object_id() const {
 #define ENCODE_FLAG_64 1 << 16
 #define ENCODE_FLAG_OBJECT_AS_ID 1 << 16
 
+// Actually, these are not flags.
+#define ENCODE_FLAG_TYPED_ARRAY_BUILTIN 1 << 16
+#define ENCODE_FLAG_TYPED_ARRAY_CLASS_NAME 2 << 16
+#define ENCODE_FLAG_TYPED_ARRAY_SCRIPT 3 << 16
+// Occupies bits 16 and 17. If we need more flags, we can place them after.
+#define ENCODE_FLAG_TYPED_ARRAY_MASK 0b11 << 16
```

Comment:
- Let me suggest putting the definitions inside parentheses.
- Isn't there an overlap between the first two and the third one? Shouldn't `ENCODE_FLAG_TYPED_ARRAY_SCRIPT` get its own bit? I think it'd be less error-prone to define them as `(1 << 16)`, `(1 << 17)` and so on.

Reply:
```suggestion
// Occupies bits 16 and 17. If we need more flags, we can place them after.
#define ENCODE_FLAG_UNTYPED_ARRAY (0b00 << 16)
#define ENCODE_FLAG_TYPED_ARRAY_BUILTIN (0b01 << 16)
#define ENCODE_FLAG_TYPED_ARRAY_CLASS_NAME (0b10 << 16)
#define ENCODE_FLAG_TYPED_ARRAY_SCRIPT (0b11 << 16)
#define ENCODE_FLAG_TYPED_ARRAY_MASK (0b11 << 16)
```

I'm not sure how to name it better.